waku/v2/node/wakunode2.go

```diff
@@ -957,3 +957,7 @@ func GetDiscv5Option(dnsDiscoveredNodes []dnsdisc.DiscoveredNode, discv5Nodes []
 func (w *WakuNode) ClusterID() uint16 {
 	return w.opts.clusterID
 }
+
+func (w *WakuNode) PeerConnector() *peermanager.PeerConnectionStrategy {
```

Comment:
Instead of exposing this specifically, i am wondering if we should have a method that indicates node connectivity change. Then this event can further be used for various actions such as stopping and starting discv5/pxClient, pausing/unpausing peer-connector and many more. 

Right now this logic is scattered across status-go and go-waku, which doesn't look clean.

Reply:
I'll try to extract this to something that can be used both in here and also in the filter subscriptions which i saw also use a similar idea.